CopyKeys/CopyKeys.ps1

```diff
@@ -1,1468 +1,1676 @@
+
+                $BekTags.DiskEncryptionKeyEncryptionKeyURL = $TargetKekUri
+                Create-Secret -Secret $EncryptedSecret.value -ContentType "Wrapped BEK"
+            }
+            else
+            {
+                Create-Secret -Secret $BekSecretBase64 -ContentType "BEK"
+            }
+
+            $CompletedList += $SourceName
+        }
+        catch
+        {
+            Write-Warning "CopyKeys not completed for $SourceName`n`n"
+            $IncompleteList[$SourceName] = $_
+        }
+    }
+
+    if ($IsKekKeyVaultNew)
+    {
+        $OutputLogger.Log(
+            $MyInvocation,
+            "Copying access policies from $($FirstKekVault.Name) to $TargetKekVault.",
+            [LogType]::INFO)
+
+        # Copying access policies to new KEK target key vault
+        $TargetKekRgName = "$($FirstKekVault.ResourceGroupName)" + "-asr"
+        Copy-AccessPolicies -TargetKeyVaultName $TargetKekVault -TargetResourceGroupName $TargetKekRgName `
+            -SourceKeyVaultName $FirstKekVault.Name -SourceAccessPolicies `
+            $FirstKekVault.Properties.AccessPolicies
+    }
+
+    if ($IsBekKeyVaultNew)
+    {
+        $OutputLogger.Log(
+            $MyInvocation,
+            "Copying access policies from $($FirstBekVault.Name) to $TargetBekVault.",
+            [LogType]::INFO)
+
+        # Copying access policies to new BEK target key vault
+        $TargetBekRgName = "$($FirstBekVault.ResourceGroupName)" + "-asr"
+        Copy-AccessPolicies -TargetKeyVaultName $TargetBekVault -TargetResourceGroupName $TargetBekRgName `
+            -SourceKeyVaultName $FirstBekVault.Name -SourceAccessPolicies `
+            $FirstBekVault.Properties.AccessPolicies
+    }
+
+    return $CompletedList
+}
+
+$ErrorActionPreference = "Stop"
+$DebugPreference = "SilentlyContinue"
+$SourceSecretsPermissions = @('get')
+$TargetSecretsPermissions = @('set')
+$SourceKeysPermissions = @('get', 'decrypt')
+$TargetKeysPermissions = @('get', 'create', 'encrypt')
+
+try
+{
+    $StartTime = Get-Date -Format 'dd-MM-yyyy-HH-mm-ss'
+    $CompletedList = @()
+    $IncompleteList = New-Object System.Collections.Hashtable
+    $OutputLogger = [Logger]::new('CopyKeys-' + $StartTime, $FilePath)
+    $OutputLogger.Log(
+        $MyInvocation,
+        "CopyKeys started - $StartTime.",
+        [LogType]::INFO)
+
+    $CompletedList = Start-CopyKeys
+}
+catch
+{
+    $UnknownError = (Out-String -InputObject $PSItem)
+    Write-Host -ForegroundColor Red -BackgroundColor Black $UnknownError
+
+    $OutputLogger.LogObject(
+        $MyInvocation,
+        $PSItem,
+        "Unknown error",
+        [LogType]::ERROR)
+}
+finally
+{
+    # Summarizes the CopyKeys status for various Vms
+    if($CompletedList.Count -gt 0)
+    {
+        $OutputLogger.Log(
+            $MyInvocation,
+            "`nCopyKeys succeeded for VMs:`n`t $($CompletedList -join "`n`t").",
+            [LogType]::OUTPUT)
+    }
+    $IncompleteList.Keys | ForEach-Object {
```

Comment:
One more request from customers is:  in the output, we should print what will be the target secret url, and target key url for each disk........it is becoming difficult to fetch this separately

Reply:
we are already displaying the secret and key both source and target related information. Have added logs to display disk name as well.